vaas-app/src/vaas/vcl/loader.py

```diff
@@ -71,9 +72,9 @@ def discard_unused_vcls(self):
                 else:
                     return_value = max(return_value, VclStatus.ERROR.value)
                     self.logger.warning("VCL %s not discarded." % vcl)
-            except AssertionError:
-                return_value = max(return_value, VclStatus.ERROR.value)
+            except AssertionError as e:
                 self.logger.warning("VCL %s not discarded." % vcl)
+                raise VclDiscardException(e)
```

Comment:
You change original behaviour - previously we can try to discard all vcl's independent of any errors - now you break everything on first problem - this is your intention ?